index.py

```diff
@@ -77,28 +84,29 @@ def build_index(in_dir, out_dict, out_postings):
             if j + 1 < num_of_blocks:
                 # do the merging process
                 # Merging block: j and j+1
-                merge_files('temp_dictionary_' + str(i) + '_' + str(j) + '.txt', 'temp_posting_'+str(i) + '_' + str(j) + '.txt',
-                            'temp_dictionary_' + str(i) + '_' + str(j+1) + '.txt', 'temp_posting_'+str(i) + '_' + str(j+1) + '.txt',
-                            'temp_dictionary_' + str(i+1) + '_' + str(k) + '.txt', 'temp_posting_'+str(i+1) + '_' + str(k) + '.txt')
+                merge_files(POSTING_DIR + 'temp_dictionary_' + str(i) + '_' + str(j) + '.txt', POSTING_DIR + 'temp_posting_'+str(i) + '_' + str(j) + '.txt',
+                            POSTING_DIR + 'temp_dictionary_' + str(i) + '_' + str(j+1) + '.txt', POSTING_DIR + 'temp_posting_'+str(i) + '_' + str(j+1) + '.txt',
+                            POSTING_DIR + 'temp_dictionary_' + str(i+1) + '_' + str(k) + '.txt', POSTING_DIR + 'temp_posting_'+str(i+1) + '_' + str(k) + '.txt')
             else:
                 # when the number is odd (left only the last data), copy the final block instead
                 # Copying block: j
-                shutil.copyfile('temp_dictionary_' + str(i) + '_' + str(j) + '.txt', 'temp_dictionary_' + str(i+1) + '_' + str(k) + '.txt')
-                shutil.copyfile('temp_posting_' + str(i) + '_' + str(j) + '.txt', 'temp_posting_' + str(i+1) + '_' + str(k) + '.txt')
-                os.remove('temp_dictionary_' + str(i) + '_' + str(j) + '.txt')
-                os.remove('temp_posting_' + str(i) + '_' + str(j) + '.txt')            
+                shutil.move(POSTING_DIR + 'temp_dictionary_' + str(i) + '_' + str(j) + '.txt', POSTING_DIR + 'temp_dictionary_' + str(i+1) + '_' + str(k) + '.txt')
+                shutil.move(POSTING_DIR + 'temp_posting_' + str(i) + '_' + str(j) + '.txt', POSTING_DIR + 'temp_posting_' + str(i+1) + '_' + str(k) + '.txt')
             k += 1
         num_of_blocks = k
 
-    # put the all posting in the dictionary and dump to a 'temporary final' file
-    shutil.copyfile('temp_posting_' + str(i+1) + '_' + str(k-1) + '.txt', 'posting_no_skip_pointer.txt')
-    shutil.copyfile('temp_dictionary_' + str(i+1) + '_' + str(k-1) + '.txt', 'dictionary_no_skip_pointer.txt')
-    os.remove('temp_posting_' + str(i+1) + '_' + str(k-1) + '.txt')
-    os.remove('temp_dictionary_' + str(i+1) + '_' + str(k-1) + '.txt')
+    try:
+        # put the all posting in the dictionary and dump to a 'temporary final' file
```

Comment:
I think this comment is misleading, something like "rename the final files" would be better.